h2/src/test/org/h2/test/unit/TestValue.java

```diff
@@ -285,8 +285,12 @@ private void testUUID() {
         ValueUuid min = ValueUuid.get(minHigh, minLow);
         assertEquals("00000000-0000-4000-8000-000000000000", min.getString());
 
-        ValueJavaObject vo = ValueJavaObject.getNoCopy(UUID.randomUUID(), null, null);
-        assertThrows(DbException.class, vo).convertTo(Value.UUID);
+        // Test conversion from ValueJavaObject to ValueUuid
+        ValueJavaObject vo_uuid = ValueJavaObject.getNoCopy(UUID.randomUUID(), null, null);
+        assertTrue(vo_uuid.convertTo(Value.UUID) instanceof ValueUuid);
```

Comment:
I guess here we have to assert that UUID you will get from ValueUuid instance will equal to the initially passed value.